com.unity.ml-agents/Editor/BehaviorParametersEditor.cs

```diff
@@ -24,25 +25,48 @@ public override void OnInspectorGUI()
             so.Update();
 
             // Drawing the Behavior Parameters
-            EditorGUI.BeginChangeCheck();
             EditorGUI.indentLevel++;
+            EditorGUI.BeginChangeCheck();
```

Comment:
Can we nest BeginChangeCheck/EndChangeChecks? This would be a little cleaner as something like
```csharp
EditorGUI.BeginChangeCheck(); // global
EditorGUI.BeginChangeCheck(); // just behaviorName
EditorGUILayout.PropertyField(so.FindProperty("m_BehaviorName"));
bool needModelUpdate = EditorGUI.EndChangeCheck();
...
var anythingChanged = EditorGUI.EndChangeCheck();
```

Reply:
i believe it's a stack, but I haven't looked at the code in a while.

Reply:
Yeah, looks like it is. I'll clean up the code to do something like this instead.